magento2-nginx/fpm-7.0/usr/local/share/magento2/install_magento_finalise.sh

```diff
@@ -2,29 +2,49 @@
 
 set -xe
 
+source /usr/local/share/bootstrap/common_functions.sh
+
 if [ -L "$0" ] ; then
     DIR="$(dirname "$(readlink -f "$0")")" ;
 else
     DIR="$(dirname "$0")" ;
-fi ;
-
-source /usr/local/share/bootstrap/common_functions.sh
+fi
 
 cd /app || exit 1
 
+set +e
+is_nfs
+IS_NFS=$?
+set -e
+
+if [ "$IS_NFS" -ne 0 ]; then
+  chown -R "${CODE_OWNER}":"${CODE_GROUP}" pub/media pub/static var
+else
+  chmod a+rw pub/media pub/static var
+fi
+
 # Preserve compiled theme files across setup:upgrade calls.
-mkdir /tmp/assets
-cp -pR pub/static/frontend/ /tmp/assets
-chown -R "${CODE_OWNER}":"${CODE_GROUP}" pub/media pub/static var
+if [ -d pub/static/frontend/ ]; then
+  mkdir /tmp/assets
```

Comment:
I wonder if something closer to the original path would be better, in the case of if a volume is used, as moving between filesystems will be slow